src/widgets/InfoBar.js

```diff
@@ -0,0 +1,202 @@
+        }
+        // Content Container Width between Icon Container and Button Container or Close Icon Container
+        // will be assigned when window will be rezied.
+        var resizeContentContainer = function () {
+            if($updateContent.length > 0 && $contentContainer.length > 0 && $updateBar.length > 0) {
+                var newWidth = $updateBar.outerWidth() - 38;
+                if($buttonContainer.length > 0) {
+                    newWidth = newWidth- $buttonContainer.outerWidth();
+                }
+                if($iconContainer.length > 0) {
+                    newWidth = newWidth - $iconContainer.outerWidth();
+                }
+                if($closeIconContainer.length > 0) {
+                    newWidth = newWidth - $closeIconContainer.outerWidth();
+                }
+
+                $contentContainer.css({
+                    "maxWidth": newWidth
+                });
+            }
+        };
+
+        resizeContentContainer();
+        $(window).on('resize.AutoUpdateBar', _.debounce(resizeContentContainer, 150));
+
+        //Event handlers on the Update Bar
+
+        // Click and key handlers on Restart button
+        if ($restart.length > 0) {
+            $restart.click(function () {
+                cleanUpdateBar();
+                exports.trigger(exports.RESTART_BTN_CLICKED);
+            });
+
+            $restart.keyup(function (event) {
+                if (event.which === SPACE_KEY) {
+                    $restart.trigger('click');
+                }
+            });
+        }
+
+        // Click and key handlers on Later button
+        if ($later.length > 0) {
+            $later.click(function () {
+                cleanUpdateBar();
+                MainViewManager.focusActivePane();
+                exports.trigger(exports.LATER_BTN_CLICKED);
+            });
+
+            $later.keyup(function (event) {
+                if (event.which === SPACE_KEY) {
+                    $later.trigger('click');
+                }
+            });
+        }
+
+        // Click and key handlers on Close button
+        if ($closeIcon.length > 0) {
+            $closeIcon.click(function () {
+                cleanUpdateBar();
+                MainViewManager.focusActivePane();
+            });
+
+            $closeIcon.keyup(function (event) {
+                if (event.which === SPACE_KEY) {
+                    $closeIcon.trigger('click');
+                }
+            });
+        }
+        $(window.document).on("keydown.AutoUpdate", function (event) {
+            var code = event.which;
+            if (code === ESC_KEY) {
+                // Keyboard input of Esc key on Update Bar dismisses and removes the bar
+                cleanUpdateBar();
+                MainViewManager.focusActivePane();
+                event.stopImmediatePropagation();
+            }
+        });
+    }
+    exports.showUpdateBar = showUpdateBar;
+    exports.RESTART_BTN_CLICKED = RESTART_BTN_CLICKED;
```

Comment:
I see this code has been copied from AutoUpdate. 

It would be better to trim down this file for just the relevant features required, or better yet make a generic InfoBar that can be reused later. Remove/rename any auto update related stuff.